book/stacks/introduction.md

```diff
@@ -5,117 +5,102 @@ collection: stacks
-distinguishes one linear structure from another is the location where
-additions and removals may occur. For example, a structure might allow
-new items to be added at only one end and removed at another; another
-may allow addition or removal from either end.
+Stacks, queues, deques, and lists are data collections with items ordered
+according to how they are added or removed. Once an item is added, it stays in
+the same position relative to its neighbors. Because of this characteristic, we
+call these collections *linear data structures*.
+
+Linear structures can be thought of as having two ends, referred to variously
+as “left” and “right”, “top” and “bottom”, or “front” and “rear”. What
+distinguishes one linear structure from another is where additions and removals
+may occur. For example, a structure might only allow new items to be added at
+one end and removed at another; another may allow addition or removal from
+either end.
 
 These variations give rise to some of the most useful data structures in
-computer science. They appear in many algorithms and can be used to
-solve a variety of important problems.
+computer science. They appear in many algorithms and can be used to solve a
+variety of important problems.
 
 Stacks
 ---
 
-A *stack* is an ordered collection of items where the addition of new
-items and the removal of existing items always takes place at the same
-end. This end is commonly referred to as the “top.” The end opposite the
-top is known as the “base.”
-
-The base of the stack is significant since items stored in the stack
-that are closer to the base represent those that have been in the stack
-the longest. The most recently added item is the one that is in position
-to be removed first. This ordering principle is sometimes called
-*LIFO*, (last in, first out). It provides an ordering based on length
-of time in the collection. Newer items are near the top, while older
-items are near the base.
-
-Many examples of stacks occur in everyday situations. Consider a stack
-of plates on a table, where it is possible to add or remove plates only
-to or from the top. Equivalently, imagine a stack of books on a desk.
-The only book whose cover is visible is the one on top. To access others
-in the stack, we need to remove the ones that are sitting on top of
-them.
+A *stack* is an ordered collection of items where the addition of new items and
+the removal of existing items always takes place at the same end. This end is
+commonly referred to as the “top”, and the opposite end is known as the “base”.
+
+Items that are closer to the base have been in the stack the longest. The most
+recently added item is always on the top of the stack and thus will be removed
+first. This ordering principle is sometimes called *LIFO* (Last In, First Out).
+It provides an ordering based on length of time in the collection. The time
+spent in the stack for any given item increases as you move from top to base.
```

Comment:
I find this phrasing a little awkward, since "the time ... increases" seems to want to refer to the passage of time itself. "Moving down" the stack doesn't affect the passage of time of course, so it's awkward reading those words together.